todo_list_app/serializers.py

```diff
@@ -27,3 +32,34 @@ def create(self, validated_data):
 
         return user
 
+
+class UserLoginSerializer(serializers.ModelSerializer):
+    email = serializers.EmailField(max_length=255, min_length=6)
+    password = serializers.CharField(max_length=68,min_length=6,write_only=True)
+    full_name = serializers.CharField(max_length=255, read_only=True)
+    access_token = serializers.CharField(max_length=255, read_only=True)
+    refresh_token = serializers.CharField(max_length=255, read_only=True)
+
+    class Meta:
+        model = User
+        fields = ['email', 'password', 'full_name', 'access_token', 'refresh_token']
+
+    def validate(self, attrs):
+        email = attrs.get('email')
+        password = attrs.get('password')
+        request = self.context.get('request')
+        user = authenticate(request, email=email, password=password)
```

Comment:
A serializer should only serialize and validate the data. Authenticating the user within `validate` is not right. A method or class should only do what it's name suggests.

Reply:
validate is authenticating the user to validate if user exists or not
it is kind of validating the user if it exists or not